icepyx/core/spatial.py

```diff
@@ -1,24 +1,34 @@
 import os
+from typing import Literal, Optional, Union, cast
 import warnings
 
 import geopandas as gpd
 import numpy as np
+from numpy.typing import NDArray
 from shapely.geometry import Polygon, box
 from shapely.geometry.polygon import orient
 
 # DevGoal: need to update the spatial_extent docstring to describe coordinate order for input
 
 
-def geodataframe(extent_type, spatial_extent, file=False, xdateline=None):
+ExtentType = Literal["bounding_box", "polygon"]
+
+
+def geodataframe(
+    extent_type: ExtentType,
+    spatial_extent: Union[str, list[float]],
+    file: bool = False,
+    xdateline: Optional[bool] = None,
+) -> gpd.GeoDataFrame:
     """
     Return a geodataframe of the spatial extent
 
     Parameters
     ----------
-    extent_type : string
+    extent_type :
         One of 'bounding_box' or 'polygon', indicating what type of input the spatial extent is
 
-    spatial_extent : string or list
+    spatial_extent :
```

Comment:
Need to update this docstring to include Polygon as an input type (since we handle it in the code!)